packages/apollo-server-core/src/plugin/schemaReporting/index.ts

```diff
@@ -152,33 +141,63 @@ export function ApolloServerPluginSchemaReporting(
           require('../../../package.json').version
         }`,
       };
+      let currentSchemaReporter: SchemaReporter | undefined;
 
-      logger.info(
-        'Apollo schema reporting starting! See your graph at ' +
-          `https://studio.apollographql.com/graph/${encodeURI(
-            graphRef,
-          )}/ with server info ${JSON.stringify(schemaReport)}`,
-      );
+      return {
+        schemaDidLoadOrUpdate({ apiSchema, coreSupergraphSdl }): void {
+          currentSchemaReporter?.stop();
 
-      const schemaReporter = new SchemaReporter({
-        schemaReport,
-        coreSchema,
-        apiKey: key,
-        endpointUrl,
-        logger,
-        // Jitter the startup between 0 and 10 seconds
-        initialReportingDelayInMs: Math.floor(
-          Math.random() * (initialDelayMaxMs ?? 10_000),
-        ),
-        fallbackReportingDelayInMs: 20_000,
-        fetcher,
-      });
+          if (overrideReportedSchema !== undefined) {
+            if (coreSupergraphSdl !== undefined) {
+              throw new Error(
+                [
+                  `The overrideReportedSchema option is incompatible with gateways`,
+                  `as the schema SDL is given directly by gateway. If you would`,
+                  `like to customize the schema SDL reported, please instead set`,
```

Comment:
This doesn't just customize the reported SDL though right? It customizes the actual schema used?

Reply:
Good catch! I've changed the code to accept `overrideReportedSchema` for gateways, interpreting it as the core schema (similar to usage reporting).